src/main/java/net/coderbot/iris/postprocess/CenterDepthSampler.java

```diff
@@ -1,62 +1,107 @@
 package net.coderbot.iris.postprocess;
 
-import net.coderbot.iris.gl.IrisRenderSystem;
+import com.google.common.collect.ImmutableSet;
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.systems.RenderSystem;
 import net.coderbot.iris.gl.framebuffer.GlFramebuffer;
-import net.coderbot.iris.rendertarget.RenderTargets;
+import net.coderbot.iris.gl.program.Program;
+import net.coderbot.iris.gl.program.ProgramBuilder;
+import net.coderbot.iris.gl.texture.DepthCopyStrategy;
+import net.coderbot.iris.gl.texture.InternalTextureFormat;
+import net.coderbot.iris.gl.texture.PixelType;
+import net.coderbot.iris.gl.uniform.UniformUpdateFrequency;
 import net.coderbot.iris.uniforms.FrameUpdateNotifier;
-import net.coderbot.iris.uniforms.transforms.SmoothedFloat;
+import net.coderbot.iris.uniforms.SystemTimeUniforms;
+import net.minecraft.client.Minecraft;
+import org.lwjgl.opengl.GL;
 import org.lwjgl.opengl.GL11C;
+import org.lwjgl.opengl.GL21C;
 
 public class CenterDepthSampler {
-	private final SmoothedFloat centerDepthSmooth;
-	private final GlFramebuffer depthBufferHolder;
-	private final RenderTargets renderTargets;
+	private static final double LN2 = Math.log(2);
 	private boolean hasFirstSample;
 	private boolean everRetrieved;
-	private FrameUpdateNotifier fakeNotifier;
+	private final Program program;
+	private final GlFramebuffer framebuffer;
+	private final int texture;
+	private final int altTexture;
+	public CenterDepthSampler(float halfLife) {
```

Comment:
please add a blank line in situations like this
```suggestion
	private final int altTexture;

	public CenterDepthSampler(float halfLife) {
```